components/suggestion/switch_channel_provider.tsx

```diff
@@ -51,28 +55,61 @@ import {isGuest} from 'mattermost-redux/utils/user_utils';
 import {Preferences} from 'mattermost-redux/constants';
 import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
 
+import {UserProfile} from '@mattermost/types/users';
+
+export type Results = {
+    matchedPretext: string;
+    terms: string[];
+    items: WrappedChannel[];
+    component: React.ElementType;
+}
+
+type ResultsCallback = (results: Results) => void;
+
 import Provider from './provider';
 import Suggestion from './suggestion.jsx';
+import { PreferenceType } from '@mattermost/types/preferences';
 
 const getState = store.getState;
 const searchProfilesMatchingWithTerm = makeSearchProfilesMatchingWithTerm();
-const ThreadsChannel = {
+const ThreadsChannel: Channel = {
     id: 'threads',
     name: 'threads',
     display_name: 'Threads',
     type: Constants.THREADS,
     delete_at: 0,
+    create_at: 0,
+    update_at: 0,
+    team_id: '',
+    header: '',
+    purpose: '',
+    last_post_at: 0,
+    last_root_post_at: 0,
+    creator_id: '',
+    scheme_id: '',
+    group_constrained: false,
 };
 
-const InsightsChannel = {
+const InsightsChannel: Channel = {
     id: 'insights',
     name: 'activity-and-insights',
     display_name: 'Insights',
     type: Constants.INSIGHTS,
     delete_at: 0,
+    create_at: 0,
```

Comment:
can we make these props optional so that we don't have to add them to the tests or here. Alternatively you can pick only the props we are using from the Channel type 
https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys